systemtest/src/main/java/io/strimzi/systemtest/utils/StUtils.java

```diff
@@ -533,6 +536,70 @@ public static void waitForKafkaTopicPartitionChange(String topicName, int partit
         );
     }
 
+    public static void waitForKafkaServiceLabelsChange(String serviceName, Map<String, String> labels) {
+        for (Map.Entry<String, String> entry : labels.entrySet()) {
+            boolean isK8sTag = entry.getKey().equals("controller-revision-hash") || entry.getKey().equals("statefulset.kubernetes.io/pod-name");
+            boolean isStrimziTag = entry.getKey().startsWith("strimzi.io/");
+            // ignoring strimzi.io and k8s labels
+            if (!(isStrimziTag || isK8sTag)) {
+                LOGGER.info("Waiting for Kafka service label change {} -> {}", entry.getKey(), entry.getValue());
+                TestUtils.waitFor("Waits for Kafka service label change " + entry.getKey() + " -> " + entry.getValue(), Constants.POLL_INTERVAL_FOR_RESOURCE_READINESS,
+                        Constants.TIMEOUT_FOR_RESOURCE_READINESS, () ->
+                                kubeClient().getService(serviceName).getMetadata().getLabels().get(entry.getKey()).equals(entry.getValue())
```

Comment:
How does this handle null pointers when the service would not exist (yet)? Will this be stable will will this make it sensitive to same race conditions etc.? (Same applies to the other checks below)

Reply:
Lambda code will fail for this check, sleep will be activated (for 1 sec) and then the check is executed again until it pass or timeout.

Reply:
Ok.